src/_pytest/pathlib.py

```diff
@@ -795,6 +787,35 @@ def resolve_pkg_root_and_module_name(
     raise CouldNotResolvePathError(f"Could not resolve for {path}")
 
 
+def _is_namespace_package(module_path: Path) -> bool:
+    # If the path has na __init__.py file, it means it is not
```

Comment:
Previous comment said

> If any of the parent paths has a __init__.py, it means it is not a namespace package (see the docs linked above).

the difference being "any", i.e. previously terminated the entire loop, now only returns `False` for the specific level but keeps trying upwards. Which is correct?

Reply:
Good catch, will check on this.